meta-lmp-base/recipes-core/initrdscripts/initramfs-framework/cryptfs

```diff
@@ -117,23 +117,49 @@ cryptfs_run() {
 			cat /run/cryptsetup/passphrase | cryptsetup luksOpen ${root_dev} ${luks_name}
 			cat /run/cryptsetup/passphrase | cryptsetup -v reencrypt --resume-only ${root_dev}
 			cryptsetup close ${luks_name}
+		else
+			# Opening at this point is not a valid operation
+			#  - We have to complete the re-encryption before opening and mounting
+                        msg "${root_dev} encryption not yet finalized, running recovery"
+                        # Check for corruption and repair if required
+                        yes "YES" | cryptsetup -v repair ${root_dev}
+                        # Resume re-encryption
+                        cryptsetup -v reencrypt --resume-only ${root_dev}
+                        # Re-encryption complete
+                        if cryptsetup luksDump ${root_dev} | grep -q "online-reencrypt"; then
+                                fatal "Unable complete unline reencryption"
+                        fi
+                        cryptsetup close ${luks_name}
 		fi
 	fi
 
-	! cat /run/cryptsetup/passphrase | cryptsetup luksOpen ${root_dev} ${luks_name} &&
-		fatal "Unable to open the LUKS partition ${root_dev}, invalid passphrase / ${luks_token} token"
+	if ! cryptsetup luksDump ${root_dev} | grep -q "systemd-${luks_token}"; then
+		msg "${luks_token} not enrolled for ${root_dev}"
+		! cat /run/cryptsetup/passphrase | cryptsetup luksOpen ${root_dev} ${luks_name} &&
+			fatal "Unable to open the LUKS partition ${root_dev}, invalid passphrase"
+	else
+		msg "${luks_token} enrolled for ${root_dev}, check if token is available"
+		cryptfs_check_token_${luks_token}
+		! cryptsetup luksOpen ${root_dev} ${luks_name} &&
+			fatal "Unable to open the LUKS partition ${root_dev}, invalid ${luks_token} token"
+	fi
 
 	e2fsck_check /dev/mapper/${luks_name}
 
+	# If not enrolled
 	if ! cryptsetup luksDump ${root_dev} | grep -q "systemd-${luks_token}"; then
 		msg "Enrolling LUKS2 keyslot based on ${luks_token} token"
 		cryptfs_enroll_${luks_token} ${root_dev}
 	fi
 
-	if ! cryptsetup luksDump ${root_dev} | grep -q "online-reencrypt"; then
+	# If enrolled
+	if cryptsetup luksDump ${root_dev} | grep -q "systemd-${luks_token}"; then
 		# CI encrypted uses keyslot 31 by default, forcing enrolled key to be 0
 		if cryptsetup luksDump ${root_dev} | grep -q "0: luks"; then
 			cryptsetup reencrypt --init-only ${root_dev}
+			if ! cryptsetup luksDump ${root_dev} | grep -q "online-reencrypt"; then
+				fatal "Unable to start online reencryption"
```

Comment:
Do we need this check here? Can reencrypt returns something valid that has to be evaluated after?